app/javascript/components/timeline-options/timeline-helper.js

```diff
@@ -0,0 +1,109 @@
+export const timelineUiOptions = (title) => ({
+  title,
+  axes: {
+    bottom: {
+      title: __('Date'),
+      mapsTo: 'date',
+      scaleType: 'time',
+    },
+    left: {
+      title: __('# of Events'),
+      mapsTo: 'value',
+      scaleType: 'linear',
+    },
+  },
+  legend: {
+    clickable: true,
+  },
+  tooltip: {
+    totalLabel: __('Total Events'),
+  },
+  points: {
+    radius: 3,
+    fillOpacity: 1,
+    filled: true,
+    enabled: true,
+  },
+  zoomBar: {
+    top: {
+      className: 'zoom-bar',
+      enabled: true,
+      type: 'graph_view',
+    },
+  },
+  height: '400px',
+});
+
+const smartAndOrStatements = (group, array) => {
+  const item = (array.length === 1) ? array[0] : `[${array.toString()}]`;
+  return `&filter[]=${group}=${item}`;
+};
+
+/** Function to build the URL to get all events that fall under the user selected parameters. */
+export const buildUrl = (values) => {
+  // TODO: verify that everyone needs / uses the host and ems on the filters
+  let url = `/api/event_streams?limit=5000&offset=0&expand=resources`;
+  url += `&attributes=group,group_level,group_name,id,event_type,message,ems_id,type,timestamp,created_on,host,source,ems_id,ext_management_system`;
```

Comment:
ems_id is here twice.

Do we need both ems_id and ext_management_system?

Do we use host?  

Maybe *_id would be better unless you really need to pull back all of the hosts, ext_management_system, etc. for all 5000 event streams.

Reply:
Note, I wonder if any plot of events would be usable with 5000 datapoints.  Since we do a limit 5000, do we page the events presented in the graph?

Reply:
Will remove the repeated ems_id, host is used below to build the [data table object](https://github.com/ManageIQ/manageiq-ui-classic/pull/8562/files/8633bc6cd50cb6efabad736213a27c0566d6cbea#diff-02c1ec6a72c05ac21c3a0899e6c958e29bf0978c967de079c0a331adfd4cb407R100). 

We can reduce the information being called with this API, to what specifically is being displayed on the table (which would remove message, ems_id, type, and ext_management_system and we can reduce host to be just host.name). We may need to reintroduce them in the future once we have a clearer picture on what needs to be displayed on the table. (i put this under next steps since this PR is getting pretty big again) 

The table doesnt paginate but because the events displayed on the table need to occur at the same time down to the second i feel it's unlikely we'll have an overwhelming amount of events per table.

Reply:
> We can reduce the information being called with this API, to what specifically is being displayed on the table (which would remove message, ems_id, type, and ext_management_system and we can reduce host to be just host.name). We may need to reintroduce them in the future once we have a clearer picture on what needs to be displayed on the table (for example, looking at timelines from vm&templates page has a 'message' row, but looking at timelines through hosts does not). I put this under next steps since this PR is getting pretty big again.
> 

Yes, that makes sense.

> The table doesnt paginate but because the events displayed on the table need to occur at the same time down to the second i feel it's unlikely we'll have an overwhelming amount of events per table.

Right, I think that's fine.  I do wonder if we should handle the situation where they select values for the type, group, level, and start and end dates and the result was still 5000(the limit).  Perhaps, a "we're showing only 5000 events, limit the date range even more to avoid 'missing' some event".

Reply:
I can include some code to let the user know that the limit has been hit 👍

Reply:
@jrafanie applied the changes